bokehjs/test/test_common/test_defaults.coffee

```diff
@@ -0,0 +1,175 @@
+      coll = Collections(name)
+      instance = new coll.model({}, {'silent' : true, 'defer_initialization' : true})
+      attrs = instance.attributes_as_json()
+      strip_ids(attrs)
+      # merge in display_defaults() which aren't in the main
+      # attributes, the overall setup here is sketchy right now
+      # because python never leaves these things unset. We need an
+      # "inherit" value, like in CSS, perhaps.
+      # But at least we can check that display_defaults() matches
+      # what Python sends.
+      display_specs = [ 'line_color', 'line_width', 'line_alpha', 'fill_color', 'fill_alpha',
+                        'text_font_size', 'text_color', 'text_alpha' ]
+      display_attr_is_spec = (name) ->
+        for s in display_specs
+          if name.endsWith(s) # gratuitous ES6
+            return true
+        return false
+      if 'display_defaults' of instance
+        display = instance.display_defaults()
+        for k in Object.keys(display)
+          if k of attrs
+            console.error("#{name}.#{k}: property present in both CoffeeScript attrs and display_defaults()")
+
+          if display_attr_is_spec(k)
+            orig = display[k]
+            if not (_.isObject(orig) and ('field' of orig or 'value' of orig))
+              # convert to 'spec dict' format
+              display[k] = { 'value' : display[k] }
+
+        attrs = _.extend({}, display, attrs)
+
+      # Merge in "factory" attrs, which don't go in attributes
+      # in the actual code, but for our current purpose
+      # let's pretend they do because they do "in effect"
+      # and we probably want to put them there eventually
+      for prop_kind, func of properties.factories
+        if prop_kind == 'visuals' # visuals is the display_defaults() case above
+          continue
+        if prop_kind of instance
+          props_of_this_kind = func(instance)
+          prop_values = {}
+          for p, v of props_of_this_kind
+            if v.field?
+              prop_values[p] = { 'field' : v.field }
+            else if v.fixed_value?
+              prop_values[p] = { 'value' : v.fixed_value }
+            else
+              n = v.value()
+              if isNaN(n) or n == null
+                prop_values[p] = null
+              else
+                prop_values[p] = { 'value' : n }
+
+            dict = prop_values[p]
+            if dict?
+              if v.units?
+                prop_values[p]['units'] = v.units
+              else
+                # properties.coffee hardcodes these units defaults
+                if prop_kind == 'distances' and 'units' not of dict
+                  dict['units'] = "data"
+                if prop_kind == 'angles' and 'units' not of dict
+                  dict['units'] = "rad"
+
+          _.extend(attrs, prop_values)
+
+      if not check_matching_defaults(name, get_defaults(name), attrs)
+        fail_count = fail_count + 1
+    expect(fail_count).to.equal 0
```

Comment:
I was actually thinking we leave out the `expect` (for now) so that the report would still print in the test, is that feasible?